### ISSUE TITLE
VizieR query_string failure

### ISSUE THREAD

**Issue by gilleslandais** [astropy/astroquery#1826]
(coming from a VizieR user)
exception is raised when query_string method is applied on a Vizier object:

eg: the following code works:
Vizier.query_object("WASP-1",radius=6*u.arcsec, catalog='WISE')

eg: the following code fails:
v = Vizier(columns=['W1mag','e_W1mag', 'W2mag', 'e_W2mag'])
v.query_object("WASP-1",radius=6*u.arcsec, catalog='WISE')

The second URL executed in VizieR returns a valid XML with records and Warning in INFO tag - what happends?
Is the VizieR VOTable well parsed by the astroquery request ?

**Comment by bsipocz** [comment]
@gilleslandais - What versions do you use? Your example code above works for me:

```
In [1]: from astroquery.vizier import Vizier                                                                            

In [2]: import astropy.units as u                                                                                       

In [3]: Vizier.query_object("WASP-1",radius=6*u.arcsec, catalog='WISE')                                                 
Out[3]: 
TableList with 3 tables:
	'0:II/311/wise' with 22 column(s) and 1 row(s) 
	'1:II/328/allwise' with 28 column(s) and 1 row(s) 
	'2:II/363/unwise' with 11 column(s) and 1 row(s) 

In [4]: v = Vizier(columns=['W1mag','e_W1mag', 'W2mag', 'e_W2mag'])                                                     

In [5]: v.query_object("WASP-1",radius=6*u.arcsec, catalog='WISE')                                                      
Out[5]: 
TableList with 2 tables:
	'0:II/311/wise' with 4 column(s) and 1 row(s) 
	'1:II/328/allwise' with 4 column(s) and 1 row(s) 
```

**Comment by gilleslandais** [comment]
I note the same issue in  Python 3.6.9 and Python.3.8.2 with astroquery (0.4.1).
I suspect that the cache hides the error? 

from astroquery.vizier import Vizier
import astropy.units as u
Vizier.query_object("WASP-1",radius=6*u.arcsec, catalog='WISE',cache=False)
v = Vizier(columns=['W1mag','e_W1mag', 'W2mag', 'e_W2mag'])
v.query_object("WASP-1",radius=6*u.arcsec, catalog='WISE', cache=False) 
 ......

(in my install, there aren't any errors if I removed the option cache=False)

**Comment by bsipocz** [comment]
clearing the cache is often a solution so I leave this open as a documentation issue, but I wonder where we could better document it. 

@gilleslandais - Would there be a place somewhere in the documentation where you would have found it if this info were there?

**Comment by gilleslandais** [comment]
and what about the  initial error ? By removing the cache, can you reproduce the problem?

**Comment by bsipocz** [comment]
I couldn't reproduce the initial problem, my working hypothesis that you had something in your cache that caused the issue in the first place.

**Comment by astromsshin** [comment]
I think their might be a problem in how the Vizier class handles the columns options. In the version 0.4.1, astroquery/vizier/core.py has the _args_to_payload function which produces the data for the post method in requests for Vizier votable CGI. 

```
if columns_default:
            body['-out'] = '*'
else:
            body['-out'] = columns_out
```

seems wrong. The columns_out is Python list object instead of string. Does it need to be changed? I have changed that part to `body['-out'] = ','.join(columns_out) ` as the same function does `body['-sort'] = ','.join(sorts_out)`. That change works fine.

```
if columns_out:
            body['-out.add'] = ','.join(columns_out)
```

part also seems problematic. All these issues may depend on how the Vizier VOTABLE CGI handles the post data.

**Comment by gilleslandais** [comment]
Hello -
it is an error following a recent security update in VizieR. Problems come indeed  from -out= which list columns in an array containing ' - I will deblock the situation before having a good practice!

**Comment by gilleslandais** [comment]
it is fixed - can you check?

regarding the URL generated by astroquery, I noted that the -out parameter (to list the expected columns) is not filled correctly - The correct usage of -out is to list columns separated by ',' (or to repeat -out for each column expected).

The current astroquery version works, because iastroquery completes the URL with  -out.add parameter (with the same column list).  -out.add parameter is dedicated for added column like position, distance which are computed by VizieR (ex: -out.add=_RA,_DE, -out.add=_r, -out.add=_t )

**Comment by astromsshin** [comment]
The change seems working although it doesn't look best to send all column parameters as arguments to the out.add parameter instead of separating out.add and out columns. I guess the design of Vizier class cannot be changed to have additional columns as a separate argument in addition to columns, considering the coherence of the class design with others.

**Comment by jwoillez** [comment]
@astromsshin, @gilleslandais, the two issues you report should be fixed by #2012. Please check if you can?
